handler/handler.go

```diff
@@ -27,20 +29,32 @@ func CreateNatsQueue(address string, port int, clientConfig NatsConfig) (*NatsQu
 	clientID := clientConfig.GetClientID()
 	clusterID := "faas-cluster"
 
-	nc, err := stan.Connect(clusterID, clientID, stan.NatsURL(natsURL))
 	queue1 := NatsQueue{
-		nc:        nc,
 		ClientID:  clientID,
 		ClusterID: clusterID,
 		NATSURL:   natsURL,
 		Topic:     "faas-request",
 	}
 
+	natsConn, err := nats.Connect(natsURL, nats.ReconnectHandler(queue1.reconnectClient(clientID, clusterID, natsURL)))
+	if err != nil {
+		return nil, err
+	}
+
+	stanConn, err := stan.Connect(clusterID, clientID, stan.NatsConn(natsConn))
```

Comment:
In current go-nats-streaming release, we have added a handler to be notified when the connection is lost. It is a bit too long to explain here, so have a look at [this](https://github.com/nats-io/go-nats-streaming#connection-status).

Note that your current reconnect handling may still be needed since once the connection is gone, you need to recreate connection and subs, but maybe you should rely on the stan connection handler instead. I would recommend not passing the NATS connection (let the stan create and own it) unless you need specific settings. Note that you can pass the URL to stan with stan.NatsURL() option.